pkg/k8sutil/k8sutil.go

```diff
@@ -599,3 +603,34 @@ func EnsureCustomGoverningService(ctx context.Context, namespace string, service
 	}
 	return nil
 }
+
+func AddStatusCleanupFinalizer(finalizers []string) ([]byte, error) {
```

Comment:
(nit) the purpose of k8sutils is to be k8s oriented so passing a metav1.Object feels more natural.
```suggestion
// AddFinalizerPatch generates the JSON patch payload which adds the finalizer to the object's metadata.
// If the finalizer is already present, it returns an empty []byte slice. 
func FinalizerAddPatch(o metav1.Object) ([]byte, error) {
```